docs/components/integration-framework/connectors/custom-built-connectors/connector-sdk.md

```diff
@@ -420,46 +455,34 @@ The secret store present in the Connector runtime environment that invokes the C
 takes care of that. Every environment can define its own way of providing such a secret store. Consult
 the [runtime environments](#runtime-environments) section to learn more about them.
 
-To replace secrets for the `connectorRequest` using the API, the object needs to implement the
-`ConnectorInput` interface. Using this interface's `replaceSecrets` method, you can use the provided
-`SecretStore` to search for and replace secrets in the properties that should support this in your
-Connector. Furthermore, the `ConnectorInput` interface comes with convenience methods to allow handling
-nested objects as well with `replaceSecretsIfNotNull`:
+To replace secrets in the `connectorRequest` using the API, you need to annotate the object's
+attributes that can hold secrets with `@Secret`. This instructs the SDK to search for and replace
+secrets in the annotated properties. You can use the `@Secret` annotation on primitive Java types
```

Comment:
```suggestion
secrets in the annotated properties. You can use the `@Secret` annotation on String fields
```

Reply:
Even better than my suggestion. Clear statement that secrets can only be contained in strings.